Files/Views/YourHome.xaml.cs

```diff
@@ -1,35 +1,63 @@
 ﻿using Files.Dialogs;
 using Files.Filesystem;
 using Files.Helpers;
+using Files.SettingsPages;
 using Files.UserControls.Widgets;
 using Files.ViewModels;
 using Files.ViewModels.Bundles;
 using Microsoft.Toolkit.Uwp;
 using System;
+using System.ComponentModel;
 using System.IO;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using Windows.ApplicationModel.AppService;
+using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
 using Windows.UI.Xaml.Navigation;
 
 namespace Files.Views
 {
-    public sealed partial class YourHome : Page, IDisposable
+    public sealed partial class YourHome : Page, INotifyPropertyChanged, IDisposable
     {
         private IShellPage AppInstance = null;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
         public SettingsViewModel AppSettings => App.AppSettings;
         public FolderSettingsViewModel FolderSettings => AppInstance?.InstanceViewModel.FolderSettings;
         public NamedPipeAsAppServiceConnection Connection => AppInstance?.ServiceConnection;
 
+        public bool NoWidgetsShown
+        {
+            get => !App.AppSettings.ShowLibraryCardsWidget && !App.AppSettings.ShowDrivesWidget && !App.AppSettings.ShowBundlesWidget && !App.AppSettings.ShowRecentFilesWidget;
+        }
+        
+        public string HomePageEmptyPageSubtitleP1Text
+        {
+            get => "HomePageEmptyPageSubtitleP1".GetLocalized().Replace(' ', (char)160) + "\ufeff";
+        }
+        
+        public string HomePageEmptyPageSubtitleP2Text
+        {
+            get => "HomePageEmptyPageSubtitleP2".GetLocalized().Replace(' ', (char)160) + "\ufeff";
```

Comment:
This had to be done, because xaml textblocks cut off last character(s) if it's space ` `

Reply:
No, `xml:preserve="space"` doesn't work.

Reply:
What is the purpose of space at the end of line?

Reply:
@tsvietOK the ` ` is for the space between text and hyperlink.

HomePageEmptyPageSubtitleP1Text + HyperlinkButton + HomePageEmptyPageSubtitleP2Text